src/__tests__/index.spec.ts

```diff
@@ -48,6 +48,34 @@ const B = /*#__PURE__*/String("b");`,
   title: "Annotated #__PURE__ for fn(fn())",
   src: `export const A = fn(fn())`,
   dest: `export const A = /*#__PURE__*/fn( /*#__PURE__*/fn());`,
+}, {
+  title: "Annotate only top level calls",
+  src: `export const A = () => { var B = call(); };`,
+  dest: `export const A = () => {
+  var B = call();
+};`,
+}, {
+  title: "Treat calls in top level IIFEs as top level calls",
+  src: `export const A = (() => { var B = call(); })()`,
+  dest: `export const A = /*#__PURE__*/(() => {
+  var B = /*#__PURE__*/call();
+})();`,
+}, {
+  title: "Skip annotating non-top level IIFEs",
+  src: `const A = () => { var B = (() => { var C = call() })(); };`,
+  dest: `const A = () => {
+  var B = (() => {
+    var C = call();
+  })();
+};`,
+}, {
+  title: "Treat nested IIFEs as top level ones",
+  src: `const A = (() => { var B = (() => { var C = call() })(); })();`,
+  dest: `const A = /*#__PURE__*/(() => {
```

Comment:
Without the top level checker, work too.

Reply:
same as https://github.com/morlay/babel-plugin-annotate-pure-call-in-variable-declarator/pull/1#discussion_r140424986